### ISSUE TITLE
it's possible to avoid check on merge commits?

### ISSUE THREAD

**Issue by thallesfreitaszup** [ZupIT/zup-dco-validator#6]
When merging a branch on another is not possible to sign commit, clicking in the `update branch` button on pull requests its also not possible to sign. Discussion on probot https://github.com/probot/dco/issues/13

**Comment by hectorcustodiozup** [comment]
Hi @thallesfreitaszup,

That is a valid point you have. I'll have to take a deeper look on the problem and see if there is a way to diferrenciate when the commit is simply a merge thus not having any new changes that should DCO verified.

Specifically about the ``update branch`` button, I can tell you so far that the github UI interface has indeed as you said little to no support for signing commits which leaves us to the manual option unfortunatelly.

Thanks, and I'll kee you up to date on the issue.

**Comment by hectorcustodiozup** [comment]
**TL;DR:** Feature could be exploited if added, and easily avoided.

After analyzing the problem it became clearer that this skip should not be added to the solution, here is why:

Each commit generates a JSON payload with its information and details, it is what we use here to detect whether or not the change is signed-off. In this structure the only thing that differentiates a simple commit from a merge commit is a message that git adds automatically.

e.g: ``Merge branch feature/branch-a into feature/branch-b``

This means that if I use that to ignore the commit, anyone could simply add this message to their commit and avoid verification even if it is not a merge.

I know it is a hassle, but I brought some alternatives and tips that may help you when commiting changes.

**To sign merge:**

You can sign merges adding the --signoff flag

	e.g: git merge feature/branch --signoff

This generates the merge commit already signed-off

![image](https://user-images.githubusercontent.com/50678762/135849958-6c89fd37-1d51-4d09-8e0f-b09a3d94b72d.png)

**To update branches**

Github UI still does not allow us to sign-off-commits automatically, so I suggest working with the command line locally for now.

You can update your branch pulling any new changes into them

	e.g: git pull origin main --signoff

![image](https://user-images.githubusercontent.com/50678762/135850139-69c52c6b-6d40-4cbf-82ff-4eda50ba7e9f.png)

**Github suggestion**

If you edit any files directly at the github interface, a message box is used to add information to the commit, you can put your signature there.

![image](https://user-images.githubusercontent.com/50678762/135850239-2abc822e-1fd2-48d3-b154-a5cac443ced9.png)


**Note: To correctly sign-off using this method I recommend you configure your github account e-email and name to the same you use locally in your git config.**

Hope these were helpful and I’ll update this thread in case something new appears but for now this issue will be closed.

**Comment by thallesfreitaszup** [comment]
@hectorcustodiozup   `In this structure the only thing that differentiates a simple commit from a merge commit is a message that git adds automatically` A merge commit have two parents, a normal commit have only one parent:(https://blog.experteer.engineering/what-are-parents-on-git-merge-commits.html). The probot dco use this difference on their implementation https://github.com/probot/dco/pull/14

**Comment by hectorcustodiozup** [comment]
I see, I did not notice that. I will reopen the issue then.

**Comment by hectorcustodiozup** [comment]
Thanks for the insight @thallesfreitaszup and sorry for my misleading info on the merge details. The feature was added to the main branch and merge commits should be skipped from now on.
